pkg/operation/care/checker.go

```diff
@@ -293,10 +293,22 @@ func (b *HealthChecker) checkManagedResourceConditions(
 ) *gardencorev1beta1.Condition {
 	if mr.Generation != mr.Status.ObservedGeneration {
 		c := b.FailedCondition(condition, gardencorev1beta1.OutdatedStatusError, fmt.Sprintf("observed generation of managed resource '%s/%s' outdated (%d/%d)", mr.Namespace, mr.Name, mr.Status.ObservedGeneration, mr.Generation))
+
+		// check if MangedResource `ResourcesApplied` condition is in failed state
+		for _, mrCondition := range mr.Status.Conditions {
+			if mrCondition.Type == resourcesv1alpha1.ResourcesApplied && mrCondition.Status == gardencorev1beta1.ConditionFalse && mrCondition.Reason == resourcesv1alpha1.ConditionApplyFailed {
+				c = b.FailedCondition(condition, mrCondition.Reason, mrCondition.Message)
+			}
+		}
+
 		return &c
 	}
 
-	for _, cond := range mr.Status.Conditions {
+	mrConditions := make([]gardencorev1beta1.Condition, len(mr.Status.Conditions))
+	copy(mrConditions, mr.Status.Conditions)
+	sortConditions(mrConditions)
```

Comment:
This is a little bit weird - if we want to check them in a specific order, then let's rather do this explicitly by using   https://github.com/gardener/gardener/blob/e753481ced744cc99e0dfb986bab54156f9bf08f/pkg/apis/core/v1beta1/helper/helper.go#L51-L61 instead of sorting and looping.